### PR TITLE
Deny undocumented unsafe block in Rust

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
Set the clippy flag `undocumented_unsafe_block` to deny. Thus it require us to document unsafe block to explain:

> Why the unsafe block is OK here.

To simplify the development, you may update your clippy configuration. On `VScode` with `rust-analyzer` extension, open `.vscode/settings.json` and add the following:

```json
"rust-analyzer.checkOnSave.command": "clippy",
"rust-analyzer.checkOnSave.extraArgs": [
    "--",
    "--deny=clippy::self_named_module_files",
    "--deny=clippy::undocumented_unsafe_blocks"
],
```

### CODE REVIEW

**Review by Volham22 — APPROVED** [review]

File: src/protocol/mod.rs [hunk 1]
```diff
@@ -148,29 +148,36 @@ macro_rules! gen_rep {
...
-                    let ret = unsafe {
-                        match rep {
-                            rep @ $mod::Rep::Ok $({ $($tt)+ })? => {
-                                let initializer =
-                                    ::pyo3::PyClassInitializer::from(([<$base_class Ok>], Self(rep)));
+                    let ret = match rep {
+                        rep @ $mod::Rep::Ok $({ $($tt)+ })? => {
+                            let initializer =
+                                ::pyo3::PyClassInitializer::from(([<$base_class Ok>], Self(rep)));
+                            // SAFETY: We use the same when creating the `initializer` and creating a new object.
```
Comment (touilleMan) [inline src/protocol/mod.rs:155]: I don't understand this comment, can you rephrase it ? (or provide more information in the current discussion)

**Comment by FirelightFlagboy** [comment]
> Can you wait my PR that add unsafe block #3219 ?

You just need to add `SAFETY:` in front of the comment you've added for the unsafe block

File: src/protocol/mod.rs (hunk 1 above)
Comment (FirelightFlagboy) [reply src/protocol/mod.rs]: I've rephrase the comment, WDYT ?

**Review by touilleMan — APPROVED** [review]
:partying_face:
